rust/crd/src/lib.rs

```diff
@@ -106,26 +131,72 @@ pub enum Error {
 #[kube(status = "TrinoClusterStatus")]
 #[serde(rename_all = "camelCase")]
 pub struct TrinoClusterSpec {
-    /// Emergency stop button, if `true` then all pods are stopped without affecting configuration (as setting `replicas` to `0` would)
+    /// Emergency stop button, if `true` then all pods are stopped without affecting configuration (as setting `replicas` to `0` would).
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub stopped: Option<bool>,
+    /// The provided trino image version in the form `xxx-stackableY.Y.Y` e.g. `387-stackable0.1.0`.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub version: Option<String>,
+    /// The discovery ConfigMap name of the Hive cluster (usually the same as the Hive cluster name).
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub hive_config_map_name: Option<String>,
+    /// The discovery ConfigMap name of the OPA cluster (usually the same as the OPA cluster name).
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub opa: Option<OpaConfig>,
-    /// A reference to a secret containing username/password for defined users
-    #[serde(default, skip_serializing_if = "Option::is_none")]
+    /// Global Trino Config for cluster settings like TLS
+    #[serde(default)]
+    pub config: GlobalTrinoConfig,
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub authentication: Option<Authentication>,
-    #[serde(default, skip_serializing_if = "Option::is_none")]
+    /// A reference to a S3 bucket.
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub s3: Option<S3ConnectionDef>,
+    /// Settings for the Coordinator Role/Process.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub coordinators: Option<Role<TrinoConfig>>,
+    /// Settings for the Worker Role/Process.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub workers: Option<Role<TrinoConfig>>,
 }
 
+#[derive(Clone, Debug, Deserialize, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GlobalTrinoConfig {
+    /// Only affects client connections.
+    /// This setting controls:
+    /// - If TLS encryption is used at all
+    /// - Which cert the servers should use to authenticate themselves against the client
+    #[serde(default = "tls_default", skip_serializing_if = "Option::is_none")]
+    pub tls: Option<TlsSecretClass>,
```

Comment:
Would it make sense to rename this `client_tls`?

Reply:
I think we called it `tls`intentionally in the ADR https://docs.stackable.tech/home/contributor/adr/ADR017-tls_authentication.html#_split_server_side_and_client_side_authentication_into_separate_config_sections
Maybe to not give users the feeling they are configuring client authentication but i don't remember precisely

Reply:
yeah i would keep it tls as well (and thats how it is named in other operators as well)